### PR TITLE
Cat reproduction

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
SUMMARY: Content "Cat Reproduction"

#### Purpose of change
Adds reproduction and kittens to mirror dog reproduction.

#### Describe the solution
As for dog, thus for cat.

#### Describe alternatives you've considered
Allowing dog owners to rule the apocalypse uncontested.

### CODE REVIEW

**Comment by wapcaplet** [comment]
How has the game not had this until now? As a cat lover, this PR pleases me greatly. Thank you for giving cats a fair chance in our apocalypse!

(Please support your local pet rescue or feral trap/neuter/release program)

**Comment by Iron-Tree** [comment]
Not really relevent, but makes me think of this: https://dwarffortresswiki.org/index.php/DF2014:Catsplosion

**Comment by Rail-Runner** [comment]
> Not really relevent, but makes me think of this: https://dwarffortresswiki.org/index.php/DF2014:Catsplosion

Shouldn't be a problem much since in this case you can freely kill and butcher excess cats.

**Comment by Iron-Tree** [comment]
I know, just started to make me laugh. I wonder if CDDA will ever take over the complexity of DF?

**Comment by Maleclypse** [comment]
And with strange aeons evens dorfs may die.

File: data/json/monsters/mammal.json [hunk 1]
```diff
@@ -149,6 +149,32 @@
...
+    "type": "MONSTER",
+    "copy-from": "mon_cat",
+    "name": "kitten",
+    "description": "A cuddly kitten meowing and scampering amidst the death.  Currently feral.",
+    "default_faction": "cat",
+    "bodytype": "dog",
+    "categories": [ "WILDLIFE" ],
+    "species": [ "MAMMAL" ],
+    "volume": "2000 ml",
+    "weight": "2 kg",
```
Comment (Night-Pryanik) [inline data/json/monsters/mammal.json]: Kittens should weigh less, I think? Around 1 kilogram, I suppose.

Comment (wapcaplet) [reply data/json/monsters/mammal.json]: Agreed - perhaps reduce the volume to 1000 ml as well? Young kittens are famous for fitting in the palm of your hand.

Comment (Night-Pryanik) [reply data/json/monsters/mammal.json]: Yeah, volume could be lowered too.

**Comment by tsulh** [comment]
> > Not really relevent, but makes me think of this: https://dwarffortresswiki.org/index.php/DF2014:Catsplosion
> 
> Shouldn't be a problem much since in this case you can freely kill and butcher excess cats.

This is awful, some people do play with immersion... gotta add the spay/neuter feature asap....

**Pull request merged**